master/internal/api_trials.go

```diff
@@ -573,6 +575,114 @@ func (a *apiServer) TrialPreemptionSignal(
 	}
 }
 
+func (a *apiServer) GetTrialSearcherTrainUntil(
+	_ context.Context, req *apiv1.GetTrialSearcherTrainUntilRequest,
+) (*apiv1.GetTrialSearcherTrainUntilResponse, error) {
+	exp, err := a.experimentActorFromTrialID(int(req.TrialId))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := a.askAtDefaultSystem(exp, trialTrainUntilReq{
+		trialID: int(req.TrialId),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	switch tResp, ok := resp.(trialTrainUntilResp); {
+	case !ok:
+		return nil, unexpectedMessageError(exp, resp)
+	case tResp.finished:
+		return nil, nil
+	default:
+		return &apiv1.GetTrialSearcherTrainUntilResponse{
+			Length: tResp.length.ToProto(),
+		}, nil
+	}
+}
+
+func (a *apiServer) ReportTrialSearcherValidation(
```

Comment:
It does feel a bit strange to not have any validation here - like, maybe it would be nice to ask for how long was trained to make sure it is correct?